### ISSUE TITLE
OSM module broken !

### ISSUE THREAD

**Issue by ebocher** [orbisgis/orbisanalysis#50]
It seems since the last OSM code refactoring, the OSM does'nt work anymore.

This real test case 

```java
@Test //disable. It uses for test purpose
    void dev() {
        H2GIS h2GIS = RANDOM_DS()
        Geometry geom = OSMTools.Utilities.getAreaFromPlace("Paimpol");

        def geomAndFilter = OSMTools.Utilities.buildGeometryAndZone(geom, -1, 0, h2GIS)

        geom = geomAndFilter.geom
        def  epsg = geom.getSRID()
        Geometry  filterArea =  geomAndFilter.filterArea

        def query = OSMTools.Utilities.buildOSMQuery(filterArea.getEnvelopeInternal(), [], OSMElement.NODE, OSMElement.WAY, OSMElement.RELATION)
        def extract = OSMTools.Loader.extract()
        if (!query.isEmpty()) {
            if (extract.execute(overpassQuery: query)) {
                def prefix = "OSM_FILE_${OSMTools.uuid}"
                def load = OSMTools.Loader.load()
                if (load(datasource: h2GIS, osmTablesPrefix: prefix, osmFilePath:extract.results.outputFilePath)) {

                    def tags = ['building']

                    def transform = OSMTools.Transform.toPolygons()
                    transform.execute(datasource: h2GIS, osmTablesPrefix: prefix, epsgCode: epsg, tags: tags)
                    assertNotNull(transform.results.outputTableName)
                    h2GIS.getTable(transform.results.outputTableName).save("/tmp/${transform.results.outputTableName}.shp")
                }
            }
        }
    }

```

returns

```java
INFO] org.orbisgis.osm.TransformTest - @ dev()
[INFO] org.orbisgis.osm.Utilities - https://nominatim.openstreetmap.org/search?q=Paimpol&limit=5&format=geojson&polygon_geojson=1
[INFO] org.orbisgis.osm.Utilities - Executing query... Paimpol
[INFO] org.orbisgis.osm.Utilities - Downloading the Nominatim data.
[INFO] org.orbisgis.osm.Loader - Extract the OSM data
[INFO] org.orbisgis.osm.Loader - https://overpass-api.de/api/interpreter?data=%5Bbbox%3A4.396723860458536E-4%2C-7.488771707149357%2C4.4023670013310003E-4%2C-7.488770662716392%5D%3B%0A%28%0A%09node%3B%0A%09way%3B%0A%09relation%3B%0A%29%3B%0A%28._%3B%3E%3B%29%3B%0Aout%3B
[INFO] org.orbisgis.osm.Loader - Executing query... [bbox:4.396723860458536E-4,-7.488771707149357,4.4023670013310003E-4,-7.488770662716392];
(
	node;
	way;
	relation;
);
(._;>;);
out;
[ERROR] org.orbisgis.osm.Loader - Cannot execute the query.
Connected as: 34220848
Current time: 2019-12-20T12:48:59Z
Rate limit: 2
2 slots available now.
Currently running queries (pid, space limit, time limit, start time):

[ERROR] org.orbisgis.osm.Loader - Cannot extract the OSM data for the query [bbox:4.396723860458536E-4,-7.488771707149357,4.4023670013310003E-4,-7.488770662716392];
(
	node;
	way;
	relation;
);
(._;>;);
out;
[INFO] org.orbisgis.osm.TransformTest - # dev()

```

I would be nice to keep in the tests some real api usages, maybe for very small areas.

**Comment by ebocher** [comment]
see https://jenkins.orbisgis.org/job/geoclimate/823/console

**Comment by SPalominos** [comment]
OSM module isn't broken, its build on Jenkins works well and this test works on my side. I'll take a look to geoclimate after resolving the issue #39 and #40.

**Comment by SPalominos** [comment]
The issue is caused by a bad detection of the SRID [here](https://github.com/orbisgis/orbisanalysis/blob/master/osm/src/main/groovy/org/orbisgis/orbisanalysis/osm/utils/Utilities.groovy#L442). So the issue seems to be linked to H2GIS and was introduced by the PR #49.
It can be avoid by giving the the processes the SRID to use.

**Comment by ebocher** [comment]
The epsg code must detected when no srid is specified. I will look for this bug.

**Comment by ebocher** [comment]
Hummm I run the real test case below and the bug appears onexecuteOverPassQuery method that returns connection.responseCode == 400

**Comment by ebocher** [comment]
Note :  my .groovy, .m2 have been deleted and the espg code is well detected.

**Comment by SPalominos** [comment]
I the PR #52 I have removed the SRID forcing and add a log for the detected SRID. you can see on Travis [here](https://travis-ci.com/orbisgis/orbisanalysis/jobs/271765969#L2795) that the SRID detected is `32630` and not `4326`.

**Comment by ebocher** [comment]
This is normal the srid used is the UTM code.
H2GIS do all geometry analysis in a carthesian plan so the epsg detected is the good one.

**Comment by SPalominos** [comment]
OK, I got the problem. The given geometry already have an SRID (4326) as it comes from the Nominatim service. But with the PR #49, the existing SRID is not taken into account and il always replace by the detected one ([here](https://github.com/orbisgis/orbisanalysis/pull/49/files#diff-3ef210aa3cb7351782bd8fe75e223163L441-R445)). So the geometry which have the SRID `4326` is set to the SRID `32630` and then transform to a geometry with the SRID `4326`, which causes the problem.
I fix that on the PR #52 and let you tell me more.

**Comment by ebocher** [comment]
Fixed
